### DISCUSSION TITLE
SSHing a running container

### DISCUSSION THREAD

**Discussion by fagiani** [buildpacks/community#58] [Q&A]
I'm running `pack` built containers on AWS's ECS with Fargate and therefore I cannot access the host machine. I'm facing a issue that only happens in production and therefore I need to be able to log into the running task to debug. 

If I use a EC2 provider as a host for the containers, I am able to log into with `docker exec -it <container-id> /bin/bash` and all ENVs will be there and I can leverage the use of `console` for instance and get rails console running (I'm using standard heroku's ruby buldpack).

As Fargate allows me to use a public IP, I've created a buildpack that will install `sshd` and create a private key so I can log into it. I am able to log in using `ssh -i <my-rsa-id>.key heorku@<container-ip>` and I will get a bash as well, but nothing is on the path and all ENVs from the Task Definition are not loaded at all...

Any clues on how to leverage whatever is done in the ENTRYPOINT or CMD of the `docker run` triggered by ECS and be able to get a behavior just like the `docker exec -it ...` command?

Any clues are highly appreciated!

**Comment by jkutner** [comment]
@fagiani Hello. You may need to set `ForceCommand bash -c "HOME=/workspace /cnb/lifecycle/launcher $your_command"` in your `sshd_config`. But in any case, you'll need to run `/cnb/lifecycle/launcher` at some point in the process, because that's what loads the buildpack environment.

I also created an [`ssh` buildpack](https://github.com/jkutner/sshd-buildpack), in case you want to share code :)

Here's an example, where I've hard code this into a buildpacks (you might also find the `web-tty` buildpack helpful):
https://github.com/jkutner/web-tty-buildpack/blob/05c0221a34dc30068bdf825ccf9d32a41ff9e199/wetty/bin/build#L50

I hope this helps.
